src/kafunk/Utility/AsyncSeq.fs

```diff
@@ -255,4 +255,39 @@ module AsyncSeq =
 
   let mergeChoice4 (s1:AsyncSeq<'a>) (s2:AsyncSeq<'b>) (s3:AsyncSeq<'c>) (s4:AsyncSeq<'d>) : AsyncSeq<Choice<'a, 'b, 'c, 'd>> =
     AsyncSeq.mergeAll 
-      [ s1 |> AsyncSeq.map Choice1Of4 ; s2 |> AsyncSeq.map Choice2Of4 ; s3 |> AsyncSeq.map Choice3Of4 ; s4 |> AsyncSeq.map Choice4Of4 ]
\ No newline at end of file
+      [ s1 |> AsyncSeq.map Choice1Of4 ; s2 |> AsyncSeq.map Choice2Of4 ; s3 |> AsyncSeq.map Choice3Of4 ; s4 |> AsyncSeq.map Choice4Of4 ]
+
+  /// merges multiple async seqs together, distribution spread across evenly throughout each async seq
+  let mergeAllRoundRobin (ss:AsyncSeq<'T> list) : AsyncSeq<'T> =
```

Comment:
Would you mind breaking this down into smaller functions with declarative names? I'm seeing 4 distinct pieces of logic, one for lines 263-269, two for 271-287 (the inner async being a separate function), and 289 to the end.